docs/tone-js.md

```diff
@@ -0,0 +1,66 @@
+
+Getting into `getSecondsAtTime(time)`. The time argument is an absolute time. It is supposed to compute what the transport time is, was, or will be at this absolute time. It does this by going through the state timeline. It looks for the last stopped event that is before the passed in time. Then it adds up the difference in the stored absolute time in events that happen after each started event. In order have an end point to this loop a temporary "paused" event is added to the state timeline. The time of this "paused" event is the time passed to `getSecondsAtTime`. While looping through the state timeline events, it also looks for offset events in `_tickOffset`. If it finds one, it resets its accumulated time to the time recorded in the offset event. This way if the user modified the transport time instead of just letting the transport play, these modifications will be accounted for. In additional to all of this there is an optimization to cache the transport time computed for a particular absolute time, this is stored in a third timeline. This way it doesn't have to fully iterate over all of the events if the same time or a future time is requested.
+
+To make this even more complex, the absolute time stored in the events and passed to `getSecondsAtTime` is the `now()` time. This includes the `lookAhead` offset. This offset is ignored because as long as the absolute time is consistent it doesn't matter. The `getSecondsTime` calculation only uses differences between these absolute times, and looking up events also uses this offset time, so it will find correct event. However where this seems to be a problem is when you want to figure out what the immediate Transport time is.
+
+### Immediate Transport time
+
+The immediate transport time is useful sometimes. For example if you want to display the current transport time in the UI. Or the sounds represent some visual items like the points this plugin is sonifying and you want to animate a line moving across the points.
+
+Tone.js provides a mechanism for drawing things synchronized with the transport using its Draw object. There is a [Animation Example](https://tonejs.github.io/examples/animationSync) which demos this. It is also described on the [Performance wiki page](https://github.com/Tonejs/Tone.js/wiki/Performance#syncing-visuals). And there is some jsdoc which doesn't show up in the generated API site in [Draw.ts](https://github.com/Tonejs/Tone.js/blob/ee43c940574c1110f11c2bf382383fdd7f73f6ed/Tone/core/util/Draw.ts#L16-L33)
+
+However this Draw approach isn't great for updating a line or time display. In those cases we want the time display to update as fast as possible. So we want to trigger an update on every animation frame. Using the Draw mechanism for this requires scheduling a loop with a interval that is less than the minium animation frame time. This could result in extra draw events being called, or not enough if the interval is wrong.
```

Comment:
```suggestion
However this Draw approach isn't great for updating a line or time display. In those cases we want the time display to update as fast as possible. So we want to trigger an update on every animation frame. Using the Draw mechanism for this requires scheduling a loop with an interval that is less than the minium animation frame time. This could result in extra draw events being called, or not enough if the interval is wrong.
```